src/item/Bow.php

```diff
@@ -110,7 +112,7 @@ public function onReleaseUsing(Player $player) : ItemUseResult{
 
 		if($player->hasFiniteResources()){
 			if(!$infinity){ //TODO: tipped arrows are still consumed when Infinity is applied
-				$player->getInventory()->removeItem($arrow);
+				$inventory?->removeItem($arrow);
```

Comment:
I don't see why a nullsafe operator is needed here?

Reply:
This is because if a player who was in creative mode at the time of 4d1f18e changes the game mode with EntityShootBowEvent, the inventory object may access the $inventory variable to which it has not been assigned. I intended to remove this as it is no longer needed in the 9524d8e but I forgot it